ykrt/src/location.rs

```diff
@@ -243,6 +269,7 @@ pub(crate) enum HotLocationKind {
     /// A trace for this HotLocation is being compiled in another trace. When compilation is
     /// complete, the compiling thread will update the state of this HotLocation.
     Compiling,
+    Counting(HotThreshold),
```

Comment:
Is the thing inside the threshold at which to give up, or the number of times we've failed to compile this location? I think it's the latter despite the type name?

Reply:
There are two "sort of counting" things:

1. "How many times have we encountered this location" is the `Counting` state
2. "How many times have we failed to trace/compile this location" is `HotLocation::tracecompilation_errors`.

In this case, while we failed fewer than "trace compilation errors threshold" we reset `Counting` to 0 and let it naturally be retraced/compiled.

In that sense, one could argue that `Counting` is somewhat misleading: it really means "how many times have we encountered this location since the count was last set to 0". But I think it's quite a nice, and very simple, back-off heuristic.

Reply:
Right, so having read a bit:
 - 1. is before the location became hot
 - 2. is a separate counter inside the hot location.

The way I'm reading this field is `CountingHavingFailed`, is that right?

I got a bit confused because I wasn't sure if this new `Counting` is the same as the other counting. I guess the answer is "kind of".

Would it be OK to add a quick docstring?

Reply:
Oops, there should have been a docstring, you're quite right! Fixed in 2a32832a.

Reply:
Thanks!